goosebit/api/v1/devices/routes.py

```diff
@@ -48,4 +48,60 @@ async def devices_delete(_: Request, config: DevicesDeleteRequest) -> StatusResp
     return StatusResponse(success=True)
 
 
+@router.patch(
+    "",
+    dependencies=[Security(validate_user_permissions, scopes=["device.write"])],
+)
+async def devices_patch(_: Request, config: DevicesPatchRequest) -> StatusResponse:
+    for uuid in config.devices:
+        updater = await get_update_manager(uuid)
+        if config.software is not None:
+            if config.software == "rollout":
+                await updater.update_update(UpdateModeEnum.ROLLOUT, None)
+            elif config.software == "latest":
+                await updater.update_update(UpdateModeEnum.LATEST, None)
+            else:
+                software = await Software.get_or_none(id=config.software)
+                await updater.update_update(UpdateModeEnum.ASSIGNED, software)
+        if config.pinned is not None:
+            await updater.update_update(UpdateModeEnum.PINNED, None)
+        if config.name is not None:
+            await updater.update_name(config.name)
+        if config.feed is not None:
+            await updater.update_feed(config.feed)
+        if config.force_update is not None:
+            await updater.update_force_update(config.force_update)
+        if config.auth_token is not None:
+            await updater.update_auth_token(config.auth_token)
+    return StatusResponse(success=True)
+
+
+@router.put(
```

Comment:
Add test case for this?

Reply:
Also rather non-standard to have such a "batch" request to update/create multiple resources. But can see the advantages if one wants to modify a big number of devices.

Reply:
More worried about getting the tests to actually run and not just break when I try to instantiate a separate app, I'm sure something to do with how I'm scoping parts of the tests, but I dont really understand well enough to fix it...

Reply:
Here I used another approach to adjust the settings during the pytest execution: https://github.com/husqvarnagroup/goosebit/pull/new/mm/device_auth